core/drivers/crypto/hisilicon/sec_main.c

```diff
@@ -0,0 +1,288 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2022-2023 HiSilicon Limited.
+ * Kunpeng hardware accelerator sec module init.
+ */
+#include "sec_main.h"
+
+#define AM_CFG_SINGLE_PORT_MAX_TRANS	0x300014
+#define SEC_CORE_INT_MASK		0x301000
+#define SEC_CORE_INT_SOURCE		0x301010
+#define SEC_RAS_CE_ENABLE		0x301050
+#define SEC_RAS_FE_ENABLE		0x301054
+#define SEC_RAS_NFE_ENABLE		0x301058
+#define SEC_MEM_START_INIT		0x301100
+#define SEC_MEM_INIT_DONE		0x301104
+#define SEC_CONTROL_REG			0x301200
+#define SEC_INTERFACE_USER_CTRL0	0x301220
+#define SEC_INTERFACE_USER_CTRL1	0x301224
+#define SEC_SAA_EN			0x301270
+#define SEC_BD_ERR_CHK_EN0		0x301380
+#define SEC_BD_ERR_CHK_EN1		0x301384
+#define SEC_BD_ERR_CHK_EN2		0x301388
+#define SEC_BD_ERR_CHK_EN3		0x30138c
+#define SEC_DYNAMIC_GATE_V3		0x30121c
+#define SEC_BD_ERR_CHK_EN2_V3		0x301508
+#define SEC_CORE_AUTO_GATE_V3		0x30212c
+#define SEC_INTERFACE_USER_CTRL0_V3	0x302220
+#define SEC_INTERFACE_USER_CTRL1_V3	0x302224
+#define SEC_SINGLE_PORT_MAX_TRANS	0x2060
+#define SEC_ABNML_INT_DISABLE		0x0
+#define SEC_RAS_CE_ENB_MASK		0x88
+#define SEC_RAS_FE_ENB_MASK		0x0
```

Comment:
nitpicking: 0 as a mask value looks strange. If that relates to some wording in the documentation, then I guess it is fine, otherwise, maybe remove `_MASK`.